### PR TITLE
Implement subtasks

### LINKED ISSUE #15: Implement Subtask
- [x] Handler
- [x] Service
- [x] GraphQL

### PR DESCRIPTION
**Pull request by B3RR10**
closes #15

### CODE REVIEW

File: src/subtask/service.rs [hunk 1]
```diff
@@ -1,8 +1,72 @@
...
+        completed_by_id: if new_subtask.completed { Some(user.id) } else { None },
+    };
+    let conn: &PooledConnection = &context.db;
+    Ok(diesel::insert_into(subtasks).values(subtask).get_result(conn)?)
+}
+
+pub(crate) fn rename(context: &Context, subtask_id: i32, new_title: String) -> ServiceResult<Subtask> {
+    use crate::schema::subtasks::dsl::{subtasks, title};
+    let _ = get_user_authenticated(context)?;
+    // TODO: Check list permissions
```
Comment (dvogt23) [inline src/subtask/service.rs]: Check the `is_owner` function in list::service https://github.com/OpenlistOrg/backend/blob/develop/src/list/service.rs#L53

File: src/subtask/service.rs [hunk 2]
```diff
@@ -1,8 +1,72 @@
...
+    let conn: &PooledConnection = &context.db;
+    Ok(diesel::update(subtasks.find(subtask_id))
+        .set(title.eq(new_title))
+        .get_result(conn)?)
+}
+
+pub(crate) fn move_to_task(context: &Context, subtask_id: i32, new_task_id: i32) -> ServiceResult<Subtask> {
+    use crate::schema::subtasks::dsl::{subtasks, task_id};
+    let _ = get_user_authenticated(context)?;
+    // TODO: Check list permissions
```
Comment (dvogt23) [inline src/subtask/service.rs]: dito

File: src/subtask/service.rs [hunk 3]
```diff
@@ -1,8 +1,72 @@
...
+    let conn: &PooledConnection = &context.db;
+    Ok(diesel::update(subtasks.find(subtask_id))
+        .set(task_id.eq(new_task_id))
+        .get_result(conn)?)
+}
+
+pub(crate) fn toggle_complete(context: &Context, subtask_id: i32) -> ServiceResult<Subtask> {
+    use crate::schema::subtasks::dsl::{completed, completed_by_id, subtasks};
+    let user = get_user_authenticated(context)?;
+    // TODO: Check list permissions
```
Comment (dvogt23) [inline src/subtask/service.rs]: dito

File: src/subtask/service.rs [hunk 4]
```diff
@@ -1,8 +1,72 @@
...
+            completed.eq(is_completed),
+            completed_by_id.eq(if is_completed { Some(user.id) } else { None }),
+        ))
+        .get_result(conn)?)
+}
+
+pub(crate) fn delete(context: &Context, subtask_id: i32) -> ServiceResult<i32> {
+    use crate::schema::subtasks::dsl::subtasks;
+    let _ = get_user_authenticated(context)?;
+    // TODO: Check list permissions
```
Comment (dvogt23) [inline src/subtask/service.rs]: dito

File: src/subtask/service.rs (hunk 1 above)
Comment (B3RR10) [reply src/subtask/service.rs]: But that only defines if the user is the owner, not if he can edit the list... I think we need a more general function that says if the user can edit the list. That's the point of the shared lists...

**Pull request merged**